lib/Doctrine/DBAL/Schema/Table.php

```diff
@@ -123,11 +123,29 @@ public function setPrimaryKey(array $columnNames, $indexName = false)
     public function addIndex(array $columnNames, $indexName = null, array $flags = [], array $options = [])
     {
         if ($indexName === null) {
-            $indexName = $this->_generateIdentifierName(
-                array_merge([$this->getName()], $columnNames),
-                'idx',
-                $this->_getMaxIdentifierLength()
-            );
+            $indexName = $this->generateNormalIndexName($columnNames);
+        }
+
+        return $this->_addIndex($this->_createIndex($columnNames, $indexName, false, false, $flags, $options));
+    }
+
+    /**
+     * @param array<int, string>   $columnNames
+     * @param array<int, string>   $flags
```

Comment:
could it be that the index name is missing?

Reply:
The coding style wants to add phpdocs if the type isn't expressive enough. An array can contain any possible type as element and a key can be one of some generics. A nullable string as a type is sufficient for the Doctrine coding style.